jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CloudFormationExplorerNodes.kt

```diff
@@ -0,0 +1,159 @@
+    private var cachedChildren: Collection<AbstractTreeNode<Any>> = if (stackStatus in FAILED_STACK_STATES || stackStatus in IN_PROGRESS_STACK_STATES) {
+        emptyList()
+    } else {
+        loadingChildren
+    }
+
+    var isChildCacheInInitialState: Boolean = true
+
+    /**
+     * Children are cached by default to prevent describeStackResources from being called each time a stack node is expanded.
+     */
+    @Suppress("UNCHECKED_CAST")
+    override fun getChildren(): Collection<AbstractTreeNode<Any>> = getChildren(false)
+
+    fun getChildren(refresh: Boolean = false): Collection<AbstractTreeNode<Any>> {
+        if (refresh) {
+            updateCachedChildren()
+        }
+
+        return cachedChildren
+    }
+
+    private fun updateCachedChildren() {
+        cachedChildren = if (stackStatus in FAILED_STACK_STATES || stackStatus in IN_PROGRESS_STACK_STATES) {
+            emptyList()
+        } else {
+            val loaded = loadServerlessStackResources()
+
+            if (loaded.isEmpty()) {
+                noResourcesChildren
+            } else {
+                loaded.filterIsInstance<AbstractTreeNode<Any>>()
+            }
+        }
+
+        isChildCacheInInitialState = false
+    }
+
+    private fun loadServerlessStackResources(): List<CloudFormationStackResourceNode> = cfnClient
+        .describeStackResources {
+            it.stackName(stackName)
+        }
+        .stackResources()
+        .filter {
+            it.resourceType() == LAMBDA_FUNCTION_TYPE && it.resourceStatus() in COMPLETE_RESOURCE_STATES
+        }
+        .map {
+            CloudFormationStackResourceNode(nodeProject, it)
+        }.toList()
+
+    override fun statusText(): String? = stackStatus.toString().toHumanReadable()
+
+    private companion object {
+        val COMPLETE_RESOURCE_STATES = setOf(ResourceStatus.CREATE_COMPLETE, ResourceStatus.UPDATE_COMPLETE)
+        val FAILED_STACK_STATES = setOf(StackStatus.CREATE_FAILED, StackStatus.DELETE_FAILED, StackStatus.ROLLBACK_FAILED)
+        val IN_PROGRESS_STACK_STATES = setOf(
+            StackStatus.CREATE_IN_PROGRESS,
+            StackStatus.DELETE_IN_PROGRESS,
+            StackStatus.ROLLBACK_IN_PROGRESS,
+            StackStatus.UPDATE_IN_PROGRESS,
+            StackStatus.UPDATE_ROLLBACK_IN_PROGRESS
+        )
+    }
+}
+
+open class CloudFormationStackResourceNode(
+    project: Project,
+    val stackResource: StackResource
+) : AwsExplorerResourceNode<StackResource>(project, LambdaClient.SERVICE_NAME, stackResource, AwsIcons.Resources.LAMBDA_FUNCTION) {
+    override fun resourceType() = "cloudformation.stack.resource"
+
+    override fun toString(): String = functionName()
+
+    override fun displayName() = functionName()
+
+    fun functionName(): String = stackResource.logicalResourceId()
```

Comment:
So it should be resourceName then?

Reply:
And we can't remote invoke these functions, but we can Lambda->Functions. I'm not sure I like this model of a resource node per type. It will balloon up of duplicated code and not consistent UX.

Reply:
I called out `CloudFormation Stack resources have no context menu at this time.` in the PR description. I did this because it seems more likely to have different types of Stack Resource Node types. Agree we'll have to find a way to improve common functionality with other nodes, like Lambda Function nodes in the future.